File: 724-find-pivot-index/724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int n=nums.size();
        vector<int> prefix(n+2,0);
        for(int i=1;i<=n;i++){
            prefix[i]=prefix[i-1]+nums[i-1];
        }
        for(int i=1;i<=n;i++){
            int leftSum=prefix[i-1];
            int rightSum=prefix[n]-prefix[i];
            
            if(leftSum==rightSum) return i-1;
        }
        return -1;
    }
};

File: 724-find-pivot-index/724-find-pivot-index.rs
impl Solution {
    pub fn pivot_index(nums: Vec<i32>) -> i32 {
        let mut sum=0;
        let n=nums.len();
        for i in 0..n{
            sum+=nums[i];
        }
        
        let mut left=0;
        
        for i in 0..n{
            let x=nums[i];
            let right=sum-left-x;
            if(left==right) {
                return i as i32;
            }
            left+=x;
        }
        return -1;
    }
}